test/CMakeLists.txt

```diff
@@ -44,6 +45,7 @@ set(GMOCK_LIBRARIES
 # Unit Test
 
 file(GLOB_RECURSE UNITTEST_FILES ${PROJECT_SOURCE_DIR}/*_unittest.cpp) 
+add_definitions(-DBUILD_UNIT_TESTS)
```

Comment:
It's unnecessary and I'll remove it. It was for friend test originally. But somehow I feel friend test is very ugly... We have to do many forward declarations of test cases which messes up the main code.